docs/modules/deploy/pages/configuring-kubernetes.adoc

```diff
@@ -341,3 +341,43 @@ For the complete example, please check link:https://guides.hazelcast.org/kuberne
+   * no data migrations are performed, speeding up the cluster shutdown
+   * no risk of out-of-memory exception due to migrations: with `ACTIVE` state and Kubernetes applying ordered shutdown of members, all data would eventually be migrated to a single member (the last one in sequence of shutdown). Therefore, it was required to plan capacity for a single member to hold all the data in the cluster or risk an out-of-memory exception.
+   * persisted cluster metadata remain consistent across all members during shutdown, allowing recovery from disk to proceed without unexpected metadata validation errors which may result in the need to perform a force-start, wiping out persistent data from one or more members.
+ - during temporary loss of members (eg during rolling restart of the cluster or in case of a pod being rescheduled by kubernetes), Hazelcast cluster switches to a configurable cluster state (`FROZEN` or `NO_MIGRATION`) to ensure speedy recovery when the member will rejoin the cluster.
+ - when scaling up or down, Hazelcast automatically switches to `ACTIVE` cluster state, so partitions are rebalanced and data is spread across all members.
+
+=== Requirements
+Automatic cluster state management requires:
+
+- Hazelcast configured with persistence enabled
+- Kubernetes discovery is in use, either explicitly configured or as auto-detected join configuration
+- Hazelcast is deployed in a `StatefulSet`
+- Hazelcast is executed with a cluster role that is allowed access to `apps` Kubernetes API group, `statefulsets` resources with `watch` verb. See https://raw.githubusercontent.com/hazelcast/hazelcast/master/kubernetes-rbac.yaml[the proposed `ClusterRole` configuration]
+
+=== Configuration
+
+Automatic cluster state management is enabled by default when Hazelcast is configured with persistence enabled and Kubernetes discovery. It can be explicitly disabled by setting Hazelcast property `hazelcast.persistence.auto.cluster.state` to `false`.
+
+Depending on the use case, setting Hazelcast property `hazelcast.persistence.auto.cluster.state.strategy` configures which cluster state will be used when members are temporarily missing from the cluster (eg pod is rescheduled or rolling restart is in progress):
+
+ - use `NO_MIGRATION` cluster state if the cluster hosts a mix of persistent and in-memory data structures or if the cluster hosts only persistent data, but you favour availability over speed of recovery. While a member is missing from the cluster, cluster state switches to `NO_MIGRATION`: this way, the first replica of partitions owned by the missing member are promoted and the data are available. When the member rejoins the cluster, persistent data can be recovered from disk and a differential sync (assuming Merkle trees are enabled, which is the case by default for persistent `IMap` and `ICache`) brings them up to speed. For in-memory data structures, a full partition sync is required.
```

Comment:
`NO_MIGRATION`: Use this as the cluster state if the cluster hosts only persistent data, but you favor availability over speed of recovery. While a member is missing from the cluster, cluster state switches to `NO_MIGRATION`. This way, the first replica of partitions owned by the missing member are promoted and the data are available. When the member rejoins the cluster, persistent data can be recovered from disk and a differential sync (assuming Merkle trees are enabled, which is the case by default for persistent `IMap` and `ICache`) brings them up to speed. For in-memory data structures, a full partition sync is required. This is the default value.